### PR TITLE
Add `@override` for files in `src/lightning/pytorch/tuner`

### PR DESCRIPTION
**Pull request by VictorPrins**
It adds the `@override` decorator for all files in `src/lightning/pytorch/tuner` (https://github.com/Lightning-AI/lightning/issues/18695).

----
:books: Documentation preview :books:: https://pytorch-lightning--19005.org.readthedocs.build/en/19005/

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
@VictorPrins What about the scheduler classes in this file? They override get_lr

File: src/lightning/pytorch/tuner/lr_finder.py [hunk 1]
```diff
@@ -456,6 +459,8 @@ def __init__(self, optimizer: torch.optim.Optimizer, end_lr: float, num_iter: in
         self.num_iter = num_iter
         super().__init__(optimizer, last_epoch)
 
+    # mypy can't follow the _TORCH_LRSCHEDULER TypeAlias, so ignore "no base method" error
+    @override  # type: ignore[misc]
     def get_lr(self) -> List[float]:
```
Comment (VictorPrins) [inline src/lightning/pytorch/tuner/lr_finder.py:464]: Good catch @awaelchli. I missed the inheritance through the `_TORCH_LRSCHEDULER` TypeAlias. So does Mypy, which raises the "no base method" error as a `misc` type error.

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
